### DISCUSSION TITLE
swig.spec rhel limitations question

### DISCUSSION THREAD

**Discussion by tomaszchrost** [swig/swig#2316] [Q&A] [answered]
Hi!

I recently had a look at the swig.spec file used when creating rpms for swig and ccache-swig, and noticed limitations when working on a rhel machines, specifically:

```
%if 0%{?rhel}
%{!?golang:%global golang 0}
%{!?octave:%global octave 0}
%{!?Rlang:%global Rlang 0}
%bcond_with build_ccache_swig
%else
%{!?octave:%global octave 1}
%{!?Rlang:%global Rlang 1}
%bcond_without build_ccache_swig
%endif
```

Why is ccache-swig not built on rhel architecture?

**Comment by wsfulton** [comment]
I don't believe this file is actually used by anyone. I think it is maintained outside of the SWIG repo by the Linux distributions. @jplesnik , Fedora has it's own version of this right, and presumably one for RHEL too? If so, I propose we delete this file from the SWIG repo.

**Reply by jplesnik** [reply]
@wsfulton We are using one spec file for Fedora and RHEL. If we want to change behavior for RHEL then using condition with macro %rhel. To be honest, I don't remember why we stop creating ccache-swig for RHEL.

**Comment by wsfulton** [comment]
The swig.spec file that SWIG ships is modified slightly by configure from https://github.com/swig/swig/blob/master/swig.spec.in. This looks nothing like the Redhat one. @jplesnik do you use the SWIG one at all, or completely ignore it? If you don't use it, I propose we remove it from the SWIG repo and stop adding it into our release tarball.

**Reply by jplesnik** [reply]
We don't use spec file shipped by SWIG.

**Reply by wsfulton** [reply, marked as answer]
I've removed swig.spec in 623b7d97a79baea923f3a64825d3afbd94c1b24d
